src/app/lists/ListsPage.tsx

```diff
@@ -5,60 +5,92 @@ import { Schema } from "@/databaseUtils";
 import supabase from "@/supabaseClient";
 import { fetchComment, fetchLists } from "@/common/fetch";
 import ListsDataView, { ListRow, listsHeaderKeysAndLabels } from "@/app/lists/ListDataview";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { subscriptionStatusRequiresErrorMessage } from "@/common/subscriptionStatusRequiresErrorMessage";
 
 interface FetchedListsData {
     data: Schema["lists"][];
     comment: string;
 }
 
-const fetchData = async (): Promise<FetchedListsData> => {
+const fetchListData = async (): Promise<FetchedListsData> => {
     const [data, comment] = await Promise.all([fetchLists(supabase), fetchComment(supabase)]);
     return { data, comment };
 };
 
+const formatListData = (listData: FetchedListsData): ListRow[] => {
+    const formattedListData = listData.data.map((row) => {
+        const newRow = {
+            primaryKey: row.primary_key,
+            rowOrder: row.row_order,
+            itemName: row.item_name,
+            ...Object.fromEntries(
+                listsHeaderKeysAndLabels
+                    .filter(([key]) => /^\d+$/.test(key))
+                    .map(([key]) => [
+                        key,
+                        {
+                            quantity: row[`quantity_for_${key}` as keyof Schema["lists"]],
+                            notes: row[`notes_for_${key}` as keyof Schema["lists"]],
+                        },
+                    ])
+            ),
+        } as ListRow; // this cast is needed here as the type system can't infer what Object.fromEntries will return
+        return newRow;
+    });
+    return formattedListData;
+};
```

Comment:
nitpick: can be simplified a little
```suggestion
const formatListData = (listData: FetchedListsData): ListRow[] =>
    listData.data.map(
        (row): ListRow =>
            ({
                primaryKey: row.primary_key,
                rowOrder: row.row_order,
                itemName: row.item_name,
                ...Object.fromEntries(
                    listsHeaderKeysAndLabels
                        // .filter(([key]) => /^\d+$/.test(key))
                        .map(([key]) => [
                            key,
                            {
                                quantity: row[`quantity_for_${key}` as keyof Schema["lists"]],
                                notes: row[`notes_for_${key}` as keyof Schema["lists"]],
                            },
                        ])
                ),
            }) as ListRow // this cast is needed here as the type system can't infer what Object.fromEntries will return
    );
```